dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -1134,6 +1165,124 @@ def dpnp_multi_dot(n, arrays, out=None):
     return result
 
 
+def dpnp_norm(x, ord=None, axis=None, keepdims=False):
+    """Compute matrix or vector norm."""
+
+    if not dpnp.issubdtype(x.dtype, dpnp.inexact):
+        x_copy = dpnp.empty_like(x, dtype=dpnp.default_float_type(x.device))
+        ht_copy_ev, _ = ti._copy_usm_ndarray_into_usm_ndarray(
+            src=dpnp.get_usm_ndarray(x),
+            dst=x_copy.get_array(),
+            sycl_queue=x.sycl_queue,
+        )
+        ht_copy_ev.wait()
+        x = x_copy
+
+    ndim = x.ndim
+    # Immediately handle some default, simple, fast, and common cases.
+    if axis is None:
+        if (
+            (ord is None)
+            or (ord in ("f", "fro") and ndim == 2)
+            or (ord == 2 and ndim == 1)
+        ):
+            # TODO: use order="K" when it is supported in dpnp.ravel
+            x = dpnp.ravel(x)
+            if dpnp.issubdtype(x.dtype, dpnp.complexfloating):
+                x_real = x.real
+                x_imag = x.imag
+                sqnorm = dpnp.dot(x_real, x_real) + dpnp.dot(x_imag, x_imag)
+            else:
+                sqnorm = dpnp.dot(x, x)
+            ret = dpnp.sqrt(sqnorm)
+            if keepdims:
+                ret = ret.reshape((1,) * ndim)
+            return ret
+
+    # Normalize the `axis` argument to a tuple.
+    if axis is None:
+        axis = tuple(range(ndim))
+    elif not isinstance(axis, tuple):
+        try:
+            axis = int(axis)
+        except Exception as e:
+            raise TypeError(
+                "'axis' must be None, an integer or a tuple of integers"
+            ) from e
+        axis = (axis,)
+
+    if len(axis) == 1:
+        axis = normalize_axis_index(axis[0], ndim)
+        if ord == dpnp.inf:
+            return dpnp.abs(x).max(axis=axis, keepdims=keepdims)
+        elif ord == -dpnp.inf:
+            return dpnp.abs(x).min(axis=axis, keepdims=keepdims)
+        elif ord == 0:
+            # Zero norm
+            # Convert to Python float in accordance with NumPy
+            return (
+                (x != 0).astype(x.real.dtype).sum(axis=axis, keepdims=keepdims)
+            )
+        elif ord == 1:
+            # special case for speedup
+            return dpnp.abs(x).sum(axis=axis, keepdims=keepdims)
+        elif ord is None or ord == 2:
+            # special case for speedup
+            s = (dpnp.conj(x) * x).real
+            return dpnp.sqrt(dpnp.sum(s, axis=axis, keepdims=keepdims))
+        else:
+            try:
+                float(ord)
+            except (TypeError, ValueError):
+                raise ValueError(f"Invalid norm order '{ord}' for vectors")
+
+            absx = dpnp.abs(x)
+            absx **= ord
```

Comment:
I modified the code in a different way. If it is acceptable, fine. otherwise I can add a explicit check for `ord` type and include a raise error when `ord` is str-type.

Reply:
Based on description of `ord` floating type is also not acceptable. I guess we should update either description or the check.